### DISCUSSION TITLE
What is the best practice to deploy xmake based binary projects with executables?

### DISCUSSION THREAD

**Discussion by qiuwei** [xmake-io/xmake#2647] [Q&A]
Suppose the target machine has compatible libc and libstdc++, what I do now is that using `xmake install --installdir=./dist`, then copy the dist directory to the target machine.

If the project depends on other packages, I copy all of the *.so  which can be found in `linkdirs` in the `on_install` hook.

I am wondering what is the best practice for deploying binaries.

**Comment by waruqi** [comment]
> If the project depends on other packages, I copy all of the *.so which can be found in linkdirs in the on_install hook.

xmake will install them automatically, you need not write on_install

**Reply by qiuwei** [reply]
Thanks for the info. Indeed if I use the packages from xrepo, the *.so are automatically copied. However, the copied shared libraries are just symlinks.
